karabo_data/tests/mockdata/detectors.py

```diff
@@ -17,22 +17,37 @@ class DetectorModule:
         'trailer',
     ]
 
-    def __init__(self, device_id, frames_per_train=64):
+    def __init__(self, device_id, frames_per_train=64, raw=True):
         self.device_id = device_id
         self.frames_per_train = frames_per_train
+        if not raw:
+            # Drop the first dim (gain) if not raw (proc data)
+            self.image_dims = self.image_dims[1:]
+        self.raw = raw
 
     def write_control(self, f):
         """Write the CONTROL and RUN data, and the relevant parts of INDEX"""
         pass
 
     @property
     def image_keys(self):
-        return [
-            ('cellId', 'u2', (1,)),
-            ('data', 'u2', self.image_dims),
-            ('length', 'u4', (1,)),
-            ('status', 'u2', (1,)),
-        ]
+        if self.raw:
+            return [
+                ('cellId', 'u2', (1,)),
+                ('data', 'u2', self.image_dims),
+                ('length', 'u4', (1,)),
+                ('status', 'u2', (1,)),
+            ]
+
+        else:
+            return [
+                ('cellId', 'u2', (1,)),
```

Comment:
Also in the example I'm looking at, `cellId` drops the extra dimension denoted by `(1,)` here.

Reply:
Ok I think I did not understand the shapes in the first place ... ('foo', 'bar', (1, )) will create an array of shape (ntrains x npulses, 1) and ('foo', 'bar', ()) will create an array of shape (ntrains x npulses)?

Reply:
Yup, that's right. The tuple gives any extra dimensions for the dataset after the first (trains x pulses) dimension.